CMakeLists.txt

```diff
@@ -101,6 +101,7 @@ set( LIBZMQPP_SOURCES
 # Staticlib
 if (ZMQPP_BUILD_STATIC)
   add_library( zmqpp-static STATIC ${LIBZMQPP_SOURCES})
+  target_compile_definitions(zmqpp-static PUBLIC -DZMQ_STATIC)
```

Comment:
`target_compile_definitions(zmqpp-static PUBLIC ZMQ_STATIC)`  will do the same but without compiler specific `-D`